packages/zero-solid/src/solid-view.ts

```diff
@@ -164,18 +180,23 @@ function materializeRelationships(change: Change): ViewChange {
   }
 }
 
-function materializeNodesRelationships(node: Node): Node {
+function materializeNodeRelationships(node: Node): Node {
   return {
     row: node.row,
     relationships: Object.fromEntries(
       Object.entries(node.relationships).map(([relationship, stream]) => {
-        const drained = [...stream()].map(materializeNodesRelationships);
-        return [relationship, () => drained];
+        const materialized = [...stream()].map(materializeNodeRelationships);
```

Comment:
This function was the one that stood out to me.

If you write in a non functional style there are no extra arrays created.

```ts
export function materializeNodeRelationships(node: Node): Node {
  const relationships: Record<string, () => Stream<Node>> = {};
  for (const relationship in node.relationships) {
    const materialized: Node[] = [];
    for (const n of node.relationships[relationship]()) {
      materialized.push(materializeNodeRelationships(n));
    }
    relationships[relationship] = () => materialized;
  }
  return {
    row: node.row,
    relationships,
  };
}
```

Reply:
doing a quick follow up for this.

Reply:
https://github.com/rocicorp/mono/pull/3703